core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -929,6 +982,56 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
         }
     }
 
+    fun signDefensivePact(duration: Int) {
+        //Note: These modifiers are additive to the freindship modifiers
+        setModifier(DiplomaticModifiers.DefensivePact, 30f)
+        otherCivDiplomacy().setModifier(DiplomaticModifiers.DefensivePact, 30f)
+        diplomaticStatus = DiplomaticStatus.DefensivePact;
+        otherCivDiplomacy().diplomaticStatus = DiplomaticStatus.DefensivePact;
+
+        for (thirdCiv in getCommonKnownCivs().filter { it.isMajorCiv() }) {
+            thirdCiv.addNotification("[${civInfo.civName}] and [$otherCivName] have signed the Defensive Pact!",
+                NotificationCategory.Diplomacy, civInfo.civName, NotificationIcon.Diplomacy, otherCivName)
+            thirdCiv.getDiplomacyManager(civInfo).setDefensivePactBasedModifier()
+        }
+        civInfo.isAtWar()
+        //Have them join our wars
```

Comment:
Are you SURE that signing a defensive pact immediately causes way?

Reply:
I think I misinterpreted an article saying "you will immediately be pulled into war" as joining previous defensive wars. I will revert this change.